### PR TITLE
feat: Implemented QR code scanning and refactored global widget (NavHeader)

### PR DESCRIPTION
**Pull request by DesmondTambe**
## Description

- Implemented QR scanner.
- Generated QR codes for users
- Share generate code as a file

Issue:

- Can't add text string to l10n generate file due to 
`"fr": 1 untranslated message(s).
To see a detailed report, use the untranslated-messages-file
option in the l10n.yaml file:
untranslated-messages-file: desiredFileName.txt
<other option>: <other selection>`

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [x] 🗑️ Chore

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read and ran all relevant commands as specified in the Running Tests section of the [Contributor Guide].
- [ ] The title of the PR follows the [Conventional Commits] guideline
- [ ] My local branch follows the naming standards in the [Deepsource Branch Naming Convention] or [Biodiversity Branch Naming Convention]
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy],
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

[Contributor Guide]: https://github.com/FlutterPlaza/.github/blob/main/CONTRIBUTING.md
[Conventional Commits]: https://www.conventionalcommits.org/en/v1.0.0-beta.4/
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[Biodiversity Branch Naming Convention]: https://bit.ly/3DyYSwM
[Deepsource Branch Naming Convention]: https://bit.ly/3Y08Gs4

### CODE REVIEW

File: ios/Runner/Info.plist [hunk 1]
```diff
@@ -1,80 +1,87 @@
...
+		<false />
+		<key>NSFaceIDUsageDescription</key>
+		<string>Uses the biometric information to authorize access to the app in place of a PIN/phone
+			number</string>
+		<key>UIApplicationSupportsIndirectInputEvents</key>
+		<true />
+		<!-- qr code scanner -->
+		<key>io.flutter.embedded_views_preview</key>
+		<true />
+		<key>NSCameraUsageDescription</key>
```
Comment (fonkamloic) [inline ios/Runner/Info.plist:83]: Please add configuration without formating the entire file.

**Review by fonkamloic — DISMISSED** [review]
Changes requested and refactor needed

**Comment by fonkamloic** [comment]
@DesmondTambe Please address the conflicts

**Comment by DesmondTambe** [comment]
> Member
Okay on it

**Comment by DesmondTambe** [comment]
@fonkamloic please the conflict is resolved, Thanks

**Pull request merged**